lib/core/networking/websockets/middleware.dart

```diff
@@ -0,0 +1,83 @@
+import '../protocol/authentication.dart';
+import '../protocol/protocol.dart';
+
+import 'session.dart';
+
+//* Type declarations
+/// Different actions middleware can handle
+enum MiddlewareAction {
+  recieve,
+  send,
+}
+
+/// A Middleware function for use in MiddlewareSession
+///
+/// Arguments: Calling [Session] instance, message, [MiddlewareAction] action and a proxy to the next [MiddlewareFunc].
+typedef MiddlewareFunc = Future<dynamic> Function(
+  dynamic,
+  dynamic,
+  MiddlewareAction,
+  Future<dynamic> Function(dynamic),
```

Comment:
This has to be `dynamic`. The documentation also clearly explains each argument.

Reply:
Yes, dynamic might be required. Still the typing provides no information about the value and therefore named parameters would resolve a problem you would otherwise have to visit the docs for.

Reply:
They are now namen in their declaration.